src/db/schema.ts

```diff
@@ -0,0 +1,330 @@
+// drizzle-orm schema definition – natural‑key version
+// ---------------------------------------------------------------
+//  Users   now use their *blockchain address* (text) as primary key
+//  SwapPools likewise use their pool address as primary key
+//  All foreign‑key columns updated accordingly
+// ---------------------------------------------------------------
+import {
+  bigint,
+  bigserial,
+  boolean,
+  integer,
+  pgEnum,
+  pgTable,
+  point,
+  primaryKey,
+  text,
+  timestamp,
+  uniqueIndex,
+  varchar,
+} from "drizzle-orm/pg-core";
+
+// ---------------------------------------------------------------
+// 1.  ENUM types
+// ---------------------------------------------------------------
+export const userRole = pgEnum("user_role", [
+  "USER",
+  "ADMIN",
+  "STAFF",
+  "SUPER_ADMIN",
+]);
+
+export const genderKind = pgEnum("gender_kind", [
+  "MALE",
+  "FEMALE",
+  "OTHER",
+  "UNSPECIFIED",
+]);
+export const commodityKind = pgEnum("commodity_kind", ["GOOD", "SERVICE"]);
+export const frequencyKind = pgEnum("frequency_kind", [
+  "ONCE",
+  "DAILY",
+  "WEEKLY",
+  "MONTHLY",
+]);
+export const reportStatus = pgEnum("report_status", [
+  "DRAFT",
+  "SUBMITTED",
+  "APPROVED",
+  "REJECTED",
+]);
+export const voucherKind = pgEnum("voucher_kind", [
+  "DEMURRAGE",
+  "ERC20",
+  "GIFTABLE",
+  "UNKNOWN",
+]);
+export const gasGiftStatus = pgEnum("gas_gift_status", [
+  "NONE",
+  "PENDING",
+  "APPROVED",
+  "DECLINED",
+]);
+export const contactKind = pgEnum("contact_kind", ["EMAIL", "PHONE", "SOCIAL"]);
+
+// ---------------------------------------------------------------
+// 2.  Core tables
+// ---------------------------------------------------------------
+export const users = pgTable("users", {
+  address: text("address").primaryKey(), // PK
+  role: userRole("role").notNull().default("USER"),
+  gasGiftStatus: gasGiftStatus("gas_gift_status").notNull().default("NONE"),
+  gasApproverAddr: text("gas_approver_addr").references(() => users.address),
```

Comment:
Do we want to enforce this self reference? I think a not null should suffice.

Reply:
Please explain a bit more. My understanding is that this will allow it to be null when a user is not approved but will be there when approved?

Reply:
@kamikazechaser

Reply:
@williamluke4 With this approach there is a caveat that the gasApproverAddr must exist in the table. But inserting the gasApproverAddr itself means some address e.g. zero address must exist for it to point to.

Reply:
> allow it to be null 

Ah right.